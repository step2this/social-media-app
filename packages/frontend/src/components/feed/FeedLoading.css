.feed-loading {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 60px 20px;
  min-height: 300px;
}

.feed-loading__spinner {
  margin-bottom: 16px;
}

.feed-loading__message {
  font-size: 16px;
  color: var(--color-text-secondary, #666);
  margin: 0;
}
