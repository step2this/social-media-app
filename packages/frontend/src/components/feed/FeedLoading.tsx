import React from 'react';
import './FeedLoading.css';

/**
 * Loading state component for feed
 * Displays a spinner and loading message
 */
export const FeedLoading: React.FC = () => {
  return (
    <div className="feed-loading" data-testid="feed-loading" aria-live="polite">
      <div className="feed-loading__spinner spinner" aria-label="Loading"></div>
      <p className="feed-loading__message">Loading your feed...</p>
    </div>
  );
};
