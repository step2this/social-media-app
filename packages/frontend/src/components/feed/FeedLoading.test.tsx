import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { FeedLoading } from './FeedLoading';

describe('FeedLoading', () => {
  it('renders loading spinner', () => {
    const { container } = render(<FeedLoading />);
    const spinner = container.querySelector('.feed-loading__spinner');
    expect(spinner).toBeInTheDocument();
  });

  it('displays "Loading your feed..." message', () => {
    render(<FeedLoading />);
    expect(screen.getByText('Loading your feed...')).toBeInTheDocument();
  });

  it('has correct CSS classes', () => {
    const { container } = render(<FeedLoading />);
    const feedLoading = container.querySelector('.feed-loading');
    expect(feedLoading).toBeInTheDocument();
    expect(feedLoading).toHaveClass('feed-loading');
  });

  it('matches snapshot', () => {
    const { container } = render(<FeedLoading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible aria-live region', () => {
    const { container } = render(<FeedLoading />);
    const feedLoading = container.querySelector('.feed-loading');
    expect(feedLoading).toHaveAttribute('aria-live', 'polite');
  });

  it('spinner has aria-label', () => {
    const { container } = render(<FeedLoading />);
    const spinner = container.querySelector('.feed-loading__spinner');
    expect(spinner).toHaveAttribute('aria-label', 'Loading');
  });

  it('container has test ID', () => {
    render(<FeedLoading />);
    expect(screen.getByTestId('feed-loading')).toBeInTheDocument();
  });

  it('renders without crashing', () => {
    const { container } = render(<FeedLoading />);
    expect(container).toBeInTheDocument();
  });
});
