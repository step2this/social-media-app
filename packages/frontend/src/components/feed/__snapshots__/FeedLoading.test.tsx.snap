// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedLoading > matches snapshot 1`] = `
<div
  aria-live="polite"
  class="feed-loading"
  data-testid="feed-loading"
>
  <div
    aria-label="Loading"
    class="feed-loading__spinner spinner"
  />
  <p
    class="feed-loading__message"
  >
    Loading your feed...
  </p>
</div>
`;
